asserts/snap_asserts_test.go

```diff
@@ -29,29 +29,98 @@ import (
 	"github.com/ubuntu-core/snappy/asserts"
 )
 
-type snapBuildSuite struct {
-	ts     time.Time
-	tsLine string
-}
-
 var (
+	_ = Suite(&snapDeclSuite{})
 	_ = Suite(&snapBuildSuite{})
 	_ = Suite(&snapRevSuite{})
 )
 
-func (sds *snapBuildSuite) SetUpSuite(c *C) {
+type snapDeclSuite struct {
+	ts     time.Time
+	tsLine string
+}
+
+func (sds *snapDeclSuite) SetUpSuite(c *C) {
 	sds.ts = time.Now().Truncate(time.Second).UTC()
 	sds.tsLine = "timestamp: " + sds.ts.Format(time.RFC3339) + "\n"
 }
 
-func (sds *snapBuildSuite) TestDecodeOK(c *C) {
+func (sds *snapDeclSuite) TestDecodeOK(c *C) {
+	encoded := "type: snap-declaration\n" +
+		"authority-id: canonical\n" +
+		"series: 16\n" +
+		"snap-id: snap-id-1\n" +
+		"snap-name: first\n" +
+		"publisher-id: dev-id1\n" +
+		"gates: snap-id-3,snap-id-4\n" +
+		sds.tsLine +
+		"body-length: 0" +
+		"\n\n" +
+		"openpgp c2ln"
+	a, err := asserts.Decode([]byte(encoded))
+	c.Assert(err, IsNil)
+	c.Check(a.Type(), Equals, asserts.SnapDeclarationType)
+	snapDecl := a.(*asserts.SnapDeclaration)
+	c.Check(snapDecl.AuthorityID(), Equals, "canonical")
+	c.Check(snapDecl.Timestamp(), Equals, sds.ts)
+	c.Check(snapDecl.Series(), Equals, "16")
+	c.Check(snapDecl.SnapID(), Equals, "snap-id-1")
+	c.Check(snapDecl.SnapName(), Equals, "first")
+	c.Check(snapDecl.PublisherID(), Equals, "dev-id1")
+	c.Check(snapDecl.Gates(), DeepEquals, []string{"snap-id-3", "snap-id-4"})
+}
+
+const (
+	snapDeclErrPrefix = "assertion snap-declaration: "
+)
+
+func (sds *snapDeclSuite) TestDecodeInvalid(c *C) {
+	encoded := "type: snap-declaration\n" +
+		"authority-id: canonical\n" +
+		"series: 16\n" +
+		"snap-id: snap-id-1\n" +
+		"snap-name: first\n" +
+		"publisher-id: dev-id1\n" +
+		"gates: snap-id-3,snap-id-4\n" +
+		sds.tsLine +
+		"body-length: 0" +
+		"\n\n" +
+		"openpgp c2ln"
+
+	invalidTests := []struct{ original, invalid, expectedErr string }{
```

Comment:
that one is part of the primary key so checking for those is not dependent on type specific code anymore, added a test anyway for completeness